src/client/modules/my/dataUpload/dataUpload.ts

```diff
@@ -0,0 +1,127 @@
+import { LightningElement } from 'lwc';
+
+export default class DataUploader extends LightningElement {
+
+    /**
+     * Retrieves the data from the input fields and makes a query to upload it to the database api.
+     */
+    gatherDataAndSend() {
+
+        let playerData = []; //data for each player input
+
+        //get entered player data
+        for(let x = 0; x < 6; x++) {
+            playerData.push({
+                "playerName": this.getValueFromInput("playerName" + (x+1).toString()),
+                "victoryPoints": parseInt(this.getValueFromInput("victoryPoints" + (x+1).toString()), 10)
+            });
+        }
+
+        //data for post
+        let data = {
+            "gameId": this.getValueFromInput("gameId"),
+            "playerData": playerData
+        };
+
+        if(this.validateInput(data)) {
+
+            //send POST request to api
+            fetch("api/v1/gameLogs", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify(data)
+            })
+            .then(response => {
+                console.log("Got response: ", response);
+                return response.json();
+            })
+            .then(returnedData => {
+                console.log("Successfully uploaded: ", returnedData);
+            })
+            .catch((error) => {
+                console.log("Error uploading data: ", error);
+            });
+
+        }
+        else {
+
+            this.template.querySelector("p[name=\"errorMessage\"]").hidden = false; //show error message
+
+        }
+
+    }
+
+    /** 
+     * Gets the value from the input field with the given name.
+     * Parameters:
+     *  name: The name of the input field in HTML.
+     * Returns:
+     *  The value currently in the input field. Can be null.
+    */
+    getValueFromInput(name) {
+        return this.template.querySelector("input[name=\"" + name + "\"]").value
+    }
+
+    /**
+     * Validates the input from the form.
+     * Parameters:
+     *  input: The object containing the input data.
+     * Returns:
+     *  True if the data is good to be sent to the database, false otherwise.
+     */
+    validateInput(input) {
```

Comment:
I think you're missing the TS return type on these functions? e.g. `validateInput(input): boolean`